### ISSUE TITLE
Newest phar doesn't work with ...$parameters

### ISSUE THREAD

**Issue by jnfingerle** [phpDocumentor/phpDocumentor2#1779]
Hi,

I've installed the newest phar provided at https://www.phpdoc.org/, self-identifying as phpdoc 2.8.5 (yet, there is an open bug report that 2.9.0 self-identifies as 2.8.5). 

Generating documentation for (minmal example)

```
<?php


class A  {
    function b($c, ...$d) {
}
```

leads to the following exception:

```
Collecting files .. OK
Initializing parser .. OK
Parsing files
Parsing <some_path>\StandardRestController.php
PHP Notice:  Undefined offset: 389 in phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/nikic/php-parser/lib/PHPParser/Lexer.php on line 121
PHP Stack trace:
PHP   1. {main}() <some_path>\phpDocumentor\phpDocumentor-2.8.5.phar:0
PHP   2. require() <some_path>\phpDocumentor\phpDocumentor-2.8.5.phar:10
PHP   3. phpDocumentor\Application->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/bin/phpdoc:23
PHP   4. Symfony\Component\Console\Application->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Application.php:183
PHP   5. Symfony\Component\Console\Application->doRun() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:126
PHP   6. Symfony\Component\Console\Application->doRunCommand() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:195
PHP   7. Symfony\Component\Console\Command\Command->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:874
PHP   8. phpDocumentor\Command\Project\RunCommand->execute() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:253
PHP   9. Symfony\Component\Console\Command\Command->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Command/Project/RunCommand.php:260
PHP  10. phpDocumentor\Parser\Command\Project\ParseCommand->execute() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:253
PHP  11. phpDocumentor\Parser\Parser->parse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/Command/Project/ParseCommand.php:224
PHP  12. phpDocumentor\Parser\Parser->parseFileIntoDescriptor() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/Parser.php:319
PHP  13. phpDocumentor\Parser\File->parse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/Parser.php:359
PHP  14. phpDocumentor\Reflection\FileReflector->process() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/File.php:66
PHP  15. phpDocumentor\Reflection\Traverser->traverse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/FileReflector.php:163
PHP  16. PHPParser_Parser->parse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/Traverser.php:52
PHP  17. phpDocumentor\Reflection\Lexer->getNextToken() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/nikic/php-parser/lib/PHPParser/Parser.php:979
PHP  18. PHPParser_Lexer->getNextToken() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/Lexer.php:48

Notice: Undefined offset: 389 in phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/nikic/php-parser/lib/PHPParser/Lexer.php on line 121

Call Stack:
    0.2187    1292512   1. {main}() <some_path>\phpDocumentor\phpDocumentor-2.8.5.phar:0
    0.2187    1286944   2. require('phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/bin/phpdoc') <some_path>\phpDocumentor\phpDocumentor-2.8.5.phar:10
    0.5156    8002368   3. phpDocumentor\Application->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/bin/phpdoc:23
    0.5156    8163344   4. Symfony\Component\Console\Application->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Application.php:183
    0.5312    8163800   5. Symfony\Component\Console\Application->doRun() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:126
    0.5312    8164272   6. Symfony\Component\Console\Application->doRunCommand() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:195
    0.5312    8164512   7. Symfony\Component\Console\Command\Command->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:874
    0.5312    8167832   8. phpDocumentor\Command\Project\RunCommand->execute() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:253
    0.5312    8197064   9. Symfony\Component\Console\Command\Command->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Command/Project/RunCommand.php:260
    0.5312    8199368  10. phpDocumentor\Parser\Command\Project\ParseCommand->execute() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:253
    0.6250    9423648  11. phpDocumentor\Parser\Parser->parse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/Command/Project/ParseCommand.php:224
    0.6250    9477520  12. phpDocumentor\Parser\Parser->parseFileIntoDescriptor() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/Parser.php:319
    0.6250    9499904  13. phpDocumentor\Parser\File->parse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/Parser.php:359
    0.6250    9594992  14. phpDocumentor\Reflection\FileReflector->process() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/File.php:66
    0.6250    9603176  15. phpDocumentor\Reflection\Traverser->traverse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/FileReflector.php:163
    0.6562   10718192  16. PHPParser_Parser->parse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/Traverser.php:52
    0.6562   10782144  17. phpDocumentor\Reflection\Lexer->getNextToken() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/nikic/php-parser/lib/PHPParser/Parser.php:979
    0.6562   10782144  18. PHPParser_Lexer->getNextToken() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/Lexer.php:48

PHP Notice:  Undefined index:  in phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/nikic/php-parser/lib/PHPParser/Parser.php on line 983
PHP Stack trace:
PHP   1. {main}() <some_path>\phpDocumentor\phpDocumentor-2.8.5.phar:0
PHP   2. require() <some_path>\phpDocumentor\phpDocumentor-2.8.5.phar:10
PHP   3. phpDocumentor\Application->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/bin/phpdoc:23
PHP   4. Symfony\Component\Console\Application->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Application.php:183
PHP   5. Symfony\Component\Console\Application->doRun() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:126
PHP   6. Symfony\Component\Console\Application->doRunCommand() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:195
PHP   7. Symfony\Component\Console\Command\Command->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Application.php:874
PHP   8. phpDocumentor\Command\Project\RunCommand->execute() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:253
PHP   9. Symfony\Component\Console\Command\Command->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Command/Project/RunCommand.php:260
PHP  10. phpDocumentor\Parser\Command\Project\ParseCommand->execute() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/symfony/console/Symfony/Component/Console/Command/Command.php:253
PHP  11. phpDocumentor\Parser\Parser->parse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/Command/Project/ParseCommand.php:224
PHP  12. phpDocumentor\Parser\Parser->parseFileIntoDescriptor() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/Parser.php:319
PHP  13. phpDocumentor\Parser\File->parse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/Parser.php:359
PHP  14. phpDocumentor\Reflection\FileReflector->process() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Parser/File.php:66
PHP  15. phpDocumentor\Reflection\Traverser->traverse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/FileReflector.php:163
PHP  16. PHPParser_Parser->parse() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/phpdocumentor/reflection/src/phpDocumentor/Reflection/Traverser.php:52

Notice: Undefined index:  in phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/vendor/nikic/php-parser/lib/PHPParser/Parser.php on line 983

Call Stack:
    0.2187    1292512   1. {main}() <some_path>\phpDocumentor\phpDocumentor-2.8.5.phar:0
    0.2187    1286944   2. require('phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/bin/phpdoc') <some_path>\phpDocumentor\phpDocumentor-2.8.5.phar:10
    0.5156    8002368   3. phpDocumentor\Application->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/bin/phpdoc:23
    0.5156    8163344   4. Symfony\Component\Console\Application->run() phar://<some_path>/phpDocumentor/phpDocumentor-2.8.5.phar/src/phpDocumentor/Application.php:183
[...]
```

**Comment by mvriel** [comment]
@jnfingerle I have just uploaded a new version of phpDocumentor.phar; can you download it and try again please?

**Comment by mvriel** [comment]
FYI: if the current release doesn't fix it, I have also some fixes in 2.9.1 related to variadics.

**Comment by jaapio** [comment]
I cannot reproduce this issue with the current version in develop. Therefor this one can be closed
